Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm

```diff
@@ -404,8 +405,11 @@ void requestPayloadForQRCode(CGImageRef image, CompletionHandler<void(NSString *
     auto queue = WorkQueue::create("com.apple.WebKit.ImageAnalysisUtilities.QRCodePayloadRequest");
     queue->dispatch([image = retainPtr(image), completion = WTFMove(completion)]() mutable {
         auto adjustedImage = imageFilledWithWhiteBackground(image.get());
-
-        auto callCompletionOnMainRunLoopWithResult = [completion = WTFMove(completion)](NSString *result) mutable {
+        auto didComplete = Box<bool>::create(false);
+        auto callCompletionOnMainRunLoopWithResult = [didComplete, completion = WTFMove(completion)](NSString *result) mutable {
+            if (*didComplete)
+                return;
+            *didComplete = true;
```

Comment:
Why can't we just use CompletionHandler's boolean operator? CompletionHandler's get nulled out on first call.